modules/caddyhttp/matchers.go

```diff
@@ -909,25 +922,40 @@ func (m MatchRemoteIP) getClientIP(r *http.Request) (net.IP, error) {
 	if err != nil {
 		ipStr = remote // OK; probably didn't have a port
 	}
+	// Some IPv6-Adresses can contain zone identifiers at the end,
+	// which are separated with "%"
+	if strings.Contains(ipStr, "%") {
+		split := strings.Split(ipStr, "%")
+		ipStr = split[0]
+		zoneID = split[1]
+	}
 	ip := net.ParseIP(ipStr)
 	if ip == nil {
-		return nil, fmt.Errorf("invalid client IP address: %s", ipStr)
+		return nil, zoneID, fmt.Errorf("invalid client IP address: %s", ipStr)
 	}
-	return ip, nil
+	return ip, zoneID, nil
 }
 
 // Match returns true if r matches m.
 func (m MatchRemoteIP) Match(r *http.Request) bool {
-	clientIP, err := m.getClientIP(r)
+	clientIP, zoneID, err := m.getClientIP(r)
 	if err != nil {
 		m.logger.Error("getting client IP", zap.Error(err))
 		return false
 	}
-	for _, ipRange := range m.cidrs {
+	zoneFilter := true
+	for i, ipRange := range m.cidrs {
 		if ipRange.Contains(clientIP) {
-			return true
+			// Check if there are zone filters assigned and if they match.
+			if m.zones[i] == "" || zoneID == m.zones[i] {
+				return true
+			}
+			zoneFilter = false
```

Comment:
I think this needs to be wrapped in a `if m.zones[i] != ""` because otherwise the log message will be written even if none of the matchers tried to match for a zone

Reply:
oh yes

Reply:
Do you really think this is necessary? If its a real IP with zone id and it does not match, what else could it be?

Reply:
Okay. I tested it with multible matchers without zone id. Got not the debug info.

Reply:
okay. you are right again ^^

Reply:
Checked it again. It seems to work without the ``if m.zones[i] != ""``.
It shows no debug info, if I have no ips with a zone. If I use a wrong one with matching IP, it shows the info.

Reply:
Yeah, you're right, sorry my brain made a silly :sweat_smile: